file_sync_down/main.py

```diff
@@ -1,39 +1,42 @@
-import os, shutil, requests, binascii, zipfile, time, hashlib
+import os, shutil, requests, binascii, zipfile, time, hashlib, json
 from . import fs_config
 
 def download_single_file(filename: str, callback):
     response = requests.get(fs_config.server_url + "/file?filename=" + filename)
-    assert response.status_code == 200
+    if response.status_code == 200:
 
-    with open(fs_config.sync_folder_location + "/" + filename, "wb+") as f:
-        f.write(response.content)
+        with open(fs_config.sync_folder_location + "/" + filename, "wb+") as f:
+            f.write(response.content)
 
-    callback(response.content)
+        callback(response.content)
 
-    # delete the file from the server
-    delete_files([filename])
+        # delete the file from the server
+        delete_files([filename])
 
 def download_zip_file(files_to_download: list):
     tmp_dir = "downloaded_zips/"
     os.makedirs(tmp_dir, exist_ok=True)
-    comma_separated_list = ','.join(files_to_download)
-    response = requests.get(fs_config.server_url + "/files?filenames=" + comma_separated_list)
-    assert response.status_code == 200
+    response = requests.post(fs_config.server_url + "/files", data=json.dumps({"filename": files_to_download}))
+    if(response.status_code == 200):
 
-    local_name = tmp_dir + binascii.b2a_hex(os.urandom(8)).decode("utf-8") + ".zip"
+        local_name = tmp_dir + binascii.b2a_hex(os.urandom(8)).decode("utf-8") + ".zip"
 
-    with open(local_name, "wb+") as f:
-        f.write(response.content)
+        with open(local_name, "wb+") as f:
+            f.write(response.content)
 
-    # delete the file(s) from the server
-    delete_files(files_to_download)
+        # delete the file(s) from the server
+        delete_files(files_to_download)
 
-    return local_name
+        return local_name
+    return []
 
 def delete_files(files_to_delete: list):
-    comma_separated_list = ','.join(files_to_delete)
-    response = requests.delete(fs_config.server_url + "/delete?filenames=" + comma_separated_list)
-    assert response.status_code == 200
+    print({'filename': files_to_delete})
+    response = requests.delete(fs_config.server_url + "/delete", data=json.dumps({"filename": files_to_delete}))
+    while response.status_code != 200:
```

Comment:
Is there a situation where this might stall indefinitely and prevent fetching new data from the server? Even if this is the only way to guarantee that the files we just downloaded get deleted on the server side, I'd argue it's more important to be ready to download new files than to clean up perfectly. Thoughts?

Could also spawn a new thread to do this request/wait loop but I'm wary of introducing more threads/processes given how many we already have

Reply:
I guess we can make it an async function to run in background, and make the delete list as a producer/consumer queue

Reply:
Yeah that could work